velox/common/memory/MallocAllocator.cpp

```diff
@@ -221,6 +221,16 @@ int64_t MallocAllocator::freeNonContiguous(Allocation& allocation) {
     Allocation::PageRun run = allocation.runAt(i);
     numFreed += run.numPages();
     void* ptr = run.data();
+    // Check if ptr was split into multiple PageRuns.
+    while (i + 1 < allocation.numRuns()) {
+      Allocation::PageRun nextRun = allocation.runAt(i + 1);
+      void* nextPtr = nextRun.data();
+      if (mallocs_.count(nextPtr) != 0) {
+        break;
+      }
+      numFreed += nextRun.numPages();
```

Comment:
Not sure if we want to add a check here that the buffers are contiguous? Thanks!

Reply:
fixed.